packages/components/src/combobox-control/test/index.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ComboboxControl from '../';
+
+const timezones = [
+	{ label: 'Greenwich Mean Time', value: 'GMT' },
+	{ label: 'Universal Coordinated Time', value: 'UTC' },
+	{ label: 'European Central Time', value: 'ECT' },
+	{ label: '(Arabic) Egypt Standard Time', value: 'ART' },
+	{ label: 'Eastern African Time', value: 'EAT' },
+	{ label: 'Middle East Time', value: 'MET' },
+	{ label: 'Near East Time', value: 'NET' },
+	{ label: 'Pakistan Lahore Time', value: 'PLT' },
+	{ label: 'India Standard Time', value: 'IST' },
+	{ label: 'Bangladesh Standard Time', value: 'BST' },
+	{ label: 'Vietnam Standard Time', value: 'VST' },
+	{ label: 'China Taiwan Time', value: 'CTT' },
+	{ label: 'Japan Standard Time', value: 'JST' },
+	{ label: 'Australia Central Time', value: 'ACT' },
+	{ label: 'Australia Eastern Time', value: 'AET' },
+	{ label: 'Solomon Standard Time', value: 'SST' },
+	{ label: 'New Zealand Standard Time', value: 'NST' },
+	{ label: 'Midway Islands Time', value: 'MIT' },
+	{ label: 'Hawaii Standard Time', value: 'HST' },
+	{ label: 'Alaska Standard Time', value: 'AST' },
+	{ label: 'Pacific Standard Time', value: 'PST' },
+	{ label: 'Phoenix Standard Time', value: 'PNT' },
+	{ label: 'Mountain Standard Time', value: 'MST' },
+	{ label: 'Central Standard Time', value: 'CST' },
+	{ label: 'Eastern Standard Time', value: 'EST' },
+	{ label: 'Indiana Eastern Standard Time', value: 'IET' },
+	{ label: 'Puerto Rico and US Virgin Islands Time', value: 'PRT' },
+	{ label: 'Canada Newfoundland Time', value: 'CNT' },
+	{ label: 'Argentina Standard Time', value: 'AGT' },
+	{ label: 'Brazil Eastern Time', value: 'BET' },
+	{ label: 'Central African Time', value: 'CAT' },
+];
+
+const defaultLabelText = 'Select a timezone';
+const getLabel = ( labelText ) => screen.getByText( labelText );
+const getInput = ( name ) => screen.getByRole( 'combobox', { name } );
+const getOption = ( name ) => screen.getByRole( 'option', { name } );
+const getAllOptions = () => screen.getAllByRole( 'option' );
+const setTargetOption = ( index ) => {
```

Comment:
Agreed on the function name and the `options` parameter. Made both of those changes! 👍 

Re: the abstraction itself... in addition to appending the `searchString` property, I also felt like it could be beneficial from a maintenance perspective to have the desired index declared once per test.

The target property gets referenced as much as three times per test. Without the abstraction, if we ever update the target of a given test we need to make sure to hit all three instances of it. With the abstraction it can just be updated once for the test in question, because the abstraction keeps it nice and DRY.

I'm happy to remove it or replace it with a `getSearchString()` if you think that would be best though.

Reply:
> Re: the abstraction itself... in addition to appending the searchString property, I also felt like it could be beneficial from a maintenance perspective to have the desired index declared once per test.
>
> The target property gets referenced as much as three times per test. Without the abstraction, if we ever update the target of a given test we need to make sure to hit all three instances of it. With the abstraction it can just be updated once for the test in question, because the abstraction keeps it nice and DRY.

For completeness sake, here's what I had in mind when writing the previous message:

<details>

<summary>click to expand</summary>

```diff
diff --git a/packages/components/src/combobox-control/test/index.js b/packages/components/src/combobox-control/test/index.js
index 2bcd980982..72033f679f 100644
--- a/packages/components/src/combobox-control/test/index.js
+++ b/packages/components/src/combobox-control/test/index.js
@@ -53,10 +53,7 @@ const getLabel = ( labelText ) => screen.getByText( labelText );
 const getInput = ( name ) => screen.getByRole( 'combobox', { name } );
 const getOption = ( name ) => screen.getByRole( 'option', { name } );
 const getAllOptions = () => screen.getAllByRole( 'option' );
-const getTargetOption = ( options, index ) => {
-	const searchString = options[ index ].label.substring( 0, 11 );
-	return { searchString, ...options[ index ] };
-};
+const getOptionSearchString = ( option ) => option.label.substring( 0, 11 );
 const setupUser = () =>
 	userEvent.setup( {
 		advanceTimers: jest.advanceTimersByTime,
@@ -138,7 +135,7 @@ describe.each( [
 
 	it( 'should select the correct option via click events', async () => {
 		const user = setupUser();
-		const targetOption = getTargetOption( timezones, 2 );
+		const targetOption = timezones[ 2 ];
 		const onChangeSpy = jest.fn();
 		render(
 			<Component
@@ -163,7 +160,7 @@ describe.each( [
 	it( 'should select the correct option via keypress events', async () => {
 		const user = setupUser();
 		const targetIndex = 4;
-		const targetOption = getTargetOption( timezones, targetIndex );
+		const targetOption = timezones[ targetIndex ];
 		const onChangeSpy = jest.fn();
 		render(
 			<Component
@@ -192,7 +189,7 @@ describe.each( [
 
 	it( 'should select the correct option from a search', async () => {
 		const user = setupUser();
-		const targetOption = getTargetOption( timezones, 13 );
+		const targetOption = timezones[ 13 ];
 		const onChangeSpy = jest.fn();
 		render(
 			<Component
@@ -207,7 +204,7 @@ describe.each( [
 		await user.tab();
 
 		// Type enough characters to ensure a predictable search result
-		await user.keyboard( targetOption.searchString );
+		await user.keyboard( getOptionSearchString( targetOption ) );
 
 		// Pressing Enter/Return selects the currently focused option
 		await user.keyboard( '{Enter}' );
@@ -219,7 +216,7 @@ describe.each( [
 
 	it( 'should render aria-live announcement upon selection', async () => {
 		const user = setupUser();
-		const targetOption = getTargetOption( timezones, 9 );
+		const targetOption = timezones[ 9 ];
 		const onChangeSpy = jest.fn();
 		render(
 			<Component
@@ -233,7 +230,7 @@ describe.each( [
 		await user.tab();
 
 		// Type enough characters to ensure a predictable search result
-		await user.keyboard( targetOption.searchString );
+		await user.keyboard( getOptionSearchString( targetOption ) );
 
 		// Pressing Enter/Return selects the currently focused option
 		await user.keyboard( '{Enter}' );
@@ -248,7 +245,7 @@ describe.each( [
 	it( 'should process multiple entries in a single session', async () => {
 		const user = setupUser();
 		const unmatchedString = 'Mordor';
-		const targetOption = getTargetOption( timezones, 6 );
+		const targetOption = timezones[ 6 ];
 		const onChangeSpy = jest.fn();
 		render(
 			<Component
@@ -289,12 +286,13 @@ describe.each( [
 		} );
 
 		// Run a second search with a valid string.
-		await user.keyboard( targetOption.searchString );
+		const searchString = getOptionSearchString( targetOption );
+		await user.keyboard( searchString );
 		const validSearchRenderedOptions = getAllOptions();
 
 		// Find option that match the search string.
 		const matches = timezones.filter( ( option ) =>
-			option.label.includes( targetOption.searchString )
+			option.label.includes( searchString )
 		);
 
 		// Confirm the rendered options match the provided dataset based on the current string.
```

</details>

Removing the `getTargetOption` abstraction doesn't necessarily mean that the `targetOption` can be saved as a variable in each test and invoked multiple times. We could then call the `getOptionSearchString` only where it's actually necessary to get a search string.

Anyway, not a big deal — we can merge this PR with or without the suggested changes

Reply:
Ah, brilliant. My concerns about repetition were more based on your suggestion of possible having no utility function at all. I really do like your proposed `getOptionSearchString()` approach, it's a lot cleaner than my previous implementation.

Adding your changes to the PR now, which should be the final piece of the puzzle for this one. Thank you @ciampo!